src/python/deepgnn/graph_engine/snark/converter/writers.py

```diff
@@ -43,26 +41,30 @@ def __init__(self, folder: str, partition: int):
         )
         self.count = 0
         self.feature_writer = NodeFeatureWriter(folder, partition)
-        self.edge_writer = EdgeWriter(folder, partition)
 
-    def add(self, node: typing.Any):
+    def add(self, node_id: int, node_type: int, features: list):
         """Write node features and data about edges from this node to binary.
 
         Args:
-            node (typing.Any): dictionary with information about
+            node_id: int
+            node_type: int
+            features: list[ndarray]
         """
-        self.nm.write(ctypes.c_uint64(node["node_id"]))  # type: ignore
-        self.nm.write(ctypes.c_uint64(self.count))  # type: ignore
-        self.nm.write(ctypes.c_int32(node["node_type"]))  # type: ignore
-        self.feature_writer.add(node)
-        self.edge_writer.add(node)
```

Comment:
This line is extremely important, node offset has to point to the specific location in the edge file. If you are separating nodes from edges, then there should be another way to keep them consistent, but I don't see it in the proposed changes.